src/server/services/location.ts

```diff
@@ -51,25 +53,31 @@ export async function createPlaceIndex(): Promise<boolean> {
     await location.createPlaceIndex(INDEX_PARAMS).promise();
     return true;
   } catch (error) {
-    logger.error(`createPlaceIndex error: ${error.message}`);
+    const typedError = error as { message: string };
+    logger.error(`createPlaceIndex error: ${typedError.message}`);
     return false;
   }
 }
 
-// TODO:- pass country into geoLocatePlaceByText so we can filter by country via AWS
 export async function geoLocatePlaceByText(
-  Text: string
+  region: string,
+  country: string
 ): Promise<Location.Types.Position> {
   if (!placeIndexExists) {
     placeIndexExists = await createPlaceIndex();
   }
 
   const location = getAWSLocationService();
+  const countryCode = region.toLowerCase().includes("vatican")
+    ? "VAT"
+    : getCountryCodeFromCountryName(country);
+
   const { Results } = await location
     .searchPlaceIndexForText({
       MaxResults: 1,
-      Text,
+      Text: region,
       IndexName: INDEX_PARAMS.IndexName,
+      FilterCountries: [countryCode],
```

Comment:
what happens if countryCode is empty which could be the case if getCountryCodeFromCountryName can't find a match